File: leetcode/python/34. Find First and Last Position of Element in Sorted Array.py
def searchRange(nums,target):
  left = 0
  right = len(nums) - 1
  res = [-1] * 2
  res[0] = -1
  has = False
  while left <= right:
    mid = left + (right-left)//2
    if nums[mid]<target:
      left+=1
    elif nums[mid] == target:
      has = True
      right-=1
    else:
      right-=1
  if has:
    res[0] = left
  left = 0
  right = len(nums) - 1
  while left <= right :
    mid = left + (right-left)//2
    if nums[mid] <= target:
      left +=1
    else:
      right -=1
    if has :
      res[1] = right
    else:
      res[1] = -1
  return res

print(searchRange([5,7,7,8,8,10],8))

File: leetcode/js/34. Find First and Last Position of Element in Sorted Array.js
var searchRange = function(nums,target){
    var left = 0
    var right = nums.length -1
    var res = []
    res[0] = -1
    var has = false
    while(left <= right){
      var mid = left + parseInt((right-left)/2)
      if(nums[mid]<target){
        left++
      }else if(nums[mid]==target){
        has = true
        right--
      }else{
        right--
      }
    }
    if(has){
      res[0] = left
    }
    left = 0
    right = nums.length - 1
    while (left <= right) {
      let mid = left + parseInt((right - left) / 2)
      if (nums[mid] <= target) {
        left++
      } else {
        right--
      }
      }
      if (has) {
        res[1] = right
      }else{
          res[1] = -1
      }
    return res
  }
  
  console.log(searchRange([5,7,7,8,8,10],8))